src/coord/src/session/vars.rs

```diff
@@ -773,6 +794,139 @@ impl Value for i32 {
     }
 }
 
+const SEC_TO_MIN: u64 = 60u64;
+const SEC_TO_HOUR: u64 = 60u64 * 60;
+const SEC_TO_DAY: u64 = 60u64 * 60 * 24;
+const MICRO_TO_MILLI: u32 = 1000u32;
+
+impl Value for Duration {
+    const TYPE_NAME: &'static str = "duration";
+
+    fn parse(s: &str) -> Result<Duration, ()> {
+        let s = s.trim();
+        // Take all numeric values from [0..]
+        let split_pos = s
+            .chars()
+            .position(|p| !char::is_numeric(p))
+            .unwrap_or_else(|| s.chars().count());
+
+        // Error if the numeric values don't parse, i.e. there aren't any.
+        let d = s[..split_pos].parse::<u64>().map_err(|_| ())?;
+
+        // We've already trimmed end
+        let (f, m): (fn(u64) -> Duration, u64) = match s[split_pos..].trim_start() {
+            "us" => (Duration::from_micros, 1),
+            // Default unit is milliseconds
+            "ms" | "" => (Duration::from_millis, 1),
+            "s" => (Duration::from_secs, 1),
+            "min" => (Duration::from_secs, SEC_TO_MIN),
+            "h" => (Duration::from_secs, SEC_TO_HOUR),
+            "d" => (Duration::from_secs, SEC_TO_DAY),
+            _ => return Err(()),
+        };
+
+        let d = if d == 0 {
```

Comment:
Why does 0 map to MAX? Is it wrong to leave 0 as 0?

Reply:
Zero means disabled, which I'm emulating by letting the timeout run for `u64::MAX` seconds.